src/TeleportJoin.sol

```diff
@@ -259,12 +266,25 @@ contract TeleportJoin {
         require(batchedDaiToFlush <= uint256(type(int256).max), "TeleportJoin/overflow");
         daiJoin.join(address(this), batchedDaiToFlush);
         if (vat.live() == 1) {
-            (, uint256 art_) = vat.urns(ilk, address(this)); // rate == RAY => normalized debt == actual debt
-            uint256 amtToPayBack = _min(batchedDaiToFlush, art_);
+            (uint256 ink_, uint256 art_) = vat.urns(ilk, address(this)); // rate == RAY => normalized debt == actual debt
+            
+            // Fix any permissionless repays that may have occurred
+            if (art_ < ink_) {
+                address vow_ = vow;
+                uint256 diff;
+                unchecked {
+                    diff = ink_ - art_;
+                }
+                vat.suck(vow_, vow_, diff * RAY); // This needs to be done to make sure we can deduct sin[vow] and vice in the next call
+                vat.grab(ilk, address(this), address(this), vow_, 0, _int256(diff)); // After this call the urn's art will be equal to its ink
```

Comment:
Is the `suck`+`grab` combo being used to avoid debt ceiling and/or dust checks? Because this can be a one-line: `vat.frob(ilk, address(this), address(this), vow_, 0, _int256(diff))`.

Reply:
Good point about GS. I think the gas savings could be pretty significant--you eliminate SSTORE operations for `vice` and `sin(vow)`, and save an external call. Ofc triggering this is probably rare in practice as DAI donations to TeleportJoin shouldn't happen much, so maybe gas isn't a big deal. The other consideration is auth--this module already needs to be a Vat `ward` because of the `slip` call, so this doesn't really increase the trust level unless we start managing authorizations on a fine-grained basis. Given all that, I don't feel too strongly. Maybe a slight bias to use `frob` for the simplicity/readability, but knowing this "always works" has a value as well.

Reply:
Yeah I think for this module makes sense using `frob`. I'll update the revert spec with this change.

Reply:
Just want to comment that maybe in an edge case the debt ceiling might give us some trouble. Let's say we want to shutdown the whole thing gracefully, then the normal thing would be setting the `ilk.line` to 0. That will provoke that anyone spamming with 1 wei repayment makes the whole `settle` to fail. We could set all the domains `line`s of this module to 0 instead, but something to take into account.

Reply:
Ok great, I will rollback the specs as well and give a final approval soon.